core/modules.rs

```diff
@@ -667,7 +674,7 @@ mod tests {
     let a_id_fut = runtime.load_module(&spec, None);
     let a_id = futures::executor::block_on(a_id_fut).expect("Failed to load");
 
-    runtime.mod_evaluate(a_id).unwrap();
+    futures::executor::block_on(runtime.mod_evaluate(a_id)).unwrap();
```

Comment:
The issue with this is the following:
Normally you can't block_on inside a block_on. Both tokio and futures crate will panic.
However the "futures" crate doesn't detect tokio's presence, so this works.
However the outer event loop may not make progress (depending on the kind of i/o, operating system etc.)

Correct me if this is not a case of nested block_on and I am mistaken.

Reply:
@piscisaureus you're right, but there's no nested `block_on`s. In `core/` we don't use `block_on` anywhere else than tests.

Reply:
I mean block_on or equivalent... [tokio_main] qualifies too I think...
You should be able to replace block_on with tokio's block_on, like [this](https://docs.rs/tokio/0.2.22/tokio/runtime/struct.Runtime.html#examples-2). If it doesn't panic we're golden.

Reply:
I'm still not sure what's in question here. 

Using `tokio` works as well:
```rust
let rt = tokio::runtime::Runtime::new().unwrap();
let a_id = rt.block_on(a_id_fut).expect("Failed to load");

rt.block_on(runtime.mod_evaluate(a_id)).unwrap();
```

Reply:
OK so no nesting. Awesome!